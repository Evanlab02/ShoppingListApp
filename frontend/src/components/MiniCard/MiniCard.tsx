// Third party imports
import { SlIcon } from "@shoelace-style/shoelace/dist/react";

// File imports
import { MiniCardProps } from "./helpers/interfaces";

// Styles
import "./styles/MiniCard.scss"

/**
 * Mini card component.
 * 
 * @param props 
 * @returns The mini card component.
 */
export default function MiniCard(props: MiniCardProps) {
    const {
        backgroundColor,
        iconBackgroundColor,
        mainText,
        subText = "",
        textColor = "#ffffff",
        iconName = "shop",
        directLink = ""
    } = props;

    return (
        <div className="mini-card" style={{ backgroundColor: backgroundColor, color: textColor }}>
            <a href={directLink}>
                <div className="mini-card-icon" style={{ backgroundColor: iconBackgroundColor }}>
                    <SlIcon name={iconName} />
                </div>
            </a>
            <div className="mini-card-content">
                <p className="value">{mainText}</p>
                <p className="sub-value">{subText}</p>
            </div>
        </div>
    );
}