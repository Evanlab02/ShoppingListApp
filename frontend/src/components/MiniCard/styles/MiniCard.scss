.mini-card {
    font-family: "Roboto", sans-serif;
    margin: 0.3rem;
    border-radius: 0.5rem;
    height: 96px;
    display: flex;
    align-items: center;

    .mini-card-icon {
        cursor: pointer;
        display: flex;
        align-items: center;
        justify-content: center;
        color: #ffffff;
        margin: 0.5rem;
        margin-left: 1rem;
        width: 50px;
        height: 50px;
        border-radius: 0.3rem;
    }

    .mini-card-icon:hover {
        opacity: 0.5;
    }

    .mini-card-content {
        .value {
            padding: 0rem;
            margin: 0rem;
            font-weight: bold;
        }

        .sub-value {
            padding: 0rem;
            margin: 0rem;
            font-weight: 400;
        }
    }
}