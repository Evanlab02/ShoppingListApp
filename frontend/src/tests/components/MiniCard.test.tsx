import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import MiniCard from "../../components/MiniCard/MiniCard";

test("Populated detail panel renders correctly.", async () => {
    const { findByTestId } = render(
        <MiniCard
            backgroundColor="#3b5fe2"
            iconBackgroundColor="#2f48a2"
            mainText="R1000"
            subText="Budget remaining"
            iconName="wallet2"
            directLink="/budgets"
        />
	);

	const card = await findByTestId("sa-mini-card");
	expect(card).toMatchSnapshot();
});