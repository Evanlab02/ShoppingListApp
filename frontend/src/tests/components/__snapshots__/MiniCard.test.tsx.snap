// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Populated detail panel renders correctly. 1`] = `
<div
  class="mini-card"
  data-testid="sa-mini-card"
  style="background-color: rgb(59, 95, 226); color: rgb(255, 255, 255);"
>
  <a
    href="/budgets"
  >
    <div
      class="mini-card-icon"
      style="background-color: rgb(47, 72, 162);"
    >
      <sl-icon
        aria-hidden="true"
        library="default"
      />
    </div>
  </a>
  <div
    class="mini-card-content"
  >
    <p
      class="value"
    >
      R1000
    </p>
    <p
      class="sub-value"
    >
      Budget remaining
    </p>
  </div>
</div>
`;
